libautoscoper/src/Camera.cpp

```diff
@@ -398,32 +376,185 @@ void Camera::loadMayaCam1(const std::string& mayacam)
     coord_frame_ = CoordFrame(&rotation[0][0], translation);
 
     // Calculate the viewport
-    viewport_[0] = -(2.0f*K[2][0] - size_[0]) / K[0][0];
-    viewport_[1] = -(2.0f*K[2][1] - size_[1]) / K[1][1];
-    viewport_[2] = 2.0f*size_[0] / K[0][0];
-    viewport_[3] = 2.0f*size_[1] / K[1][1];
+    calculateViewport(K[2][0], K[2][1], K[0][0], K[1][1]);
 
+    // Calculate the image plane
+    double z = -0.5* (K[0][0] + K[1][1]);
+    calculateImagePlane(K[2][0], K[2][1], z);
+  }
 
-    // Choose the scaling factor such that the image plane will be on the
-    // other side of the origin from the camera. The values in the mayacam
-    // file are discarded.
-    double z = - 0.5* (K[0][0] + K[1][1]);
-    double distance = sqrt(translation[0] * translation[0] +
-      translation[1] * translation[1] +
-      translation[2] * translation[2]);
-    double scale = -1.5*distance / z;
+  void Camera::loadVTKCam1(const std::string& filename) {
+    // Open and prase the file
+    double version = -1.0, view_angle = -1.0, image_width = -1.0 , image_height = -1.0;
+    double focal_point[3] = {-1.0, -1.0, -1.0}, camera_position[3] = { -1.0, -1.0, -1.0 }, view_up[3] = { -1.0, -1.0, -1.0 };
+    std::fstream file(filename.c_str(), std::ios::in);
+    if (!file.is_open()) {
+      throw std::runtime_error("Error opening VTKCam file: " + filename);
+    }
+    std::string line;
+    // The file is a series of key value pairs separated by a colon, # denotes a comment
+    int line_num = 1;
+    while (safeGetline(file, line)) {
+      // Ignore comments and empty lines
+      if (line.empty() || line[0] == '#') {
+        line_num++;
+        continue;
+      }
+      // Split the line into key and value
+      std::string key, value;
+      std::istringstream line_stream(line);
+      if (!getline(line_stream, key, ':')) {
+        file.close();
+        throw std::runtime_error(vtkCamReadingError("1", line_num, filename, "Error parsing key."));
+      }
+      if (!getline(line_stream, value, ':')) {
+        file.close();
+        throw std::runtime_error(vtkCamReadingError("1", line_num, filename, "Error parsing value."));
+      }
+      // Parse the key value pair
+      if (key == "version") {
+        std::istringstream value_stream(value);
+        if (!(value_stream >> version)) {
+          file.close();
+          throw std::runtime_error(vtkCamReadingError("1", line_num, filename, "Error parsing version number."));
+        }
+      }
```

Comment:
The cameras used to render the VRGs are defined as VTK cameras within Slicer. I had initially done the conversion from VTK -> Autoscoper cameras within Autoscoper to increase the ease of testing when making small tweaks. 

I would be cautious about using the `MayaCam2` since there is some reorganization of the file within Autoscoper and while this is documented and something we can account for. I feel this will make debugging the camera positions moving forward more difficult. 

https://github.com/BrownBiomechanics/Autoscoper/blob/01e4e9333362781acf98e955f8ef67df7985fb2e/libautoscoper/src/Camera.cpp#L338-L368

Reply:
Last, anticipating the integration with Slicer where we already have support for reading/writing json based document, would you consider using "json" based format instead ? 

By doing so, we would loose the ability for having comment in the file but would: (1) potentially remove the need for manually writing/saving the file, (2) adopt a format consistent with the one already used in Slicer[^1][^2][^3] and (3) capitalize on expertise writing schema[^4][^5]

> I feel this will make debugging the camera positions moving forward more difficult

Alternatively, if we add a comment to the documentation stating that the format is for internal use only and may change without notices, I would have no problem keeping the current implementation.

[^1]: https://slicer.readthedocs.io/en/latest/user_guide/data_loading_and_saving.html#markups
[^2]: https://slicer.readthedocs.io/en/latest/user_guide/modules/terminologies.html#how-to
[^3]: https://github.com/Slicer/Slicer/blob/main/Modules/Loadable/Volumes/Resources/VolumeDisplayPresets.json
[^4]: https://github.com/Slicer/Slicer/tree/main/Modules/Loadable/Volumes/Resources/Schema
[^5]: https://github.com/Slicer/Slicer/tree/main/Modules/Loadable/Markups/Resources/Schema